data_tools/firebase_client_wrapper.py

```diff
@@ -115,6 +138,21 @@ def set_messages_content_batch(dataset_id, messages, batch_size=500):
 
     print ("Written {} messages".format(i))
 
+def set_dataset_messages_content_batch(dataset_id, messages, batch_size=500):
+    next_batch_to_write = []
+    current_shard_size = len(get_shard_messages(dataset_id, get_shard_count(dataset_id)))
+    for message in messages:
+        if current_shard_size >= MAX_SHARD_SIZE:
```

Comment:
This isn't how shards work, what this does is creates a growing number of size limited buckets, rather than a fixed number of shards that divides the content up between them. All uses of 'shard' will need to be replaced with something like 'subset'.

I can see the reasoning for doing it this way rather than in shards, there are some disadvantages

Reply:
Oh I had assumed that must also be a valid interpretation. The stricter definition of sharding from CS didn't make sense, as it doesn't fit into the RA workflow so well (they're already having to open and scroll to the bottom of ~10 datasets per day some days), and dynamically adjusting the number of shards as projects grow is scary. It also seemed like unnecessary additional engineering effort for a temporary fix on a tight timescale.